drivers/r2dbc-driver/src/main/kotlin/app/cash/sqldelight/driver/r2dbc/R2dbcDriver.kt

```diff
@@ -12,8 +12,16 @@ import kotlinx.coroutines.flow.toList
 import kotlinx.coroutines.reactive.asFlow
 import kotlinx.coroutines.reactive.awaitFirstOrNull
 import kotlinx.coroutines.reactive.awaitSingle
-
-class R2dbcDriver(val connection: Connection) : SqlDriver {
+import org.reactivestreams.Subscriber
+import org.reactivestreams.Subscription
+
+class R2dbcDriver(
+  val connection: Connection,
+  /**
+   * This callback is called after [close]. It either contains an error or null, representing a successful close.
+   */
+  val closed: (Throwable?) -> Unit,
```

Comment:
```suggestion
  val closed: (Throwable?) -> Unit = {},
```
I feel like in most cases we don't actually need any special handing for this.

Reply:
Hm, yes and no. If you don't provide this value, you don't have an option to wait for the close call, because `driver.close()` will return immediately.

Reply:
Of course, this call isn't suspending or blocking, see the tests for a waiting implementation.
So yes, you still need to implement the listener (or we could provide an `CoroutineScope` overload).

Reply:
Sounds good to me either way